src/hotspot/share/oops/stackChunkOop.inline.hpp

```diff
@@ -135,8 +135,8 @@ inline bool stackChunkOopDesc::is_in_chunk(void* p) const {
 }
 
 bool stackChunkOopDesc::is_usable_in_chunk(void* p) const {
-#if (defined(X86) || defined(AARCH64)) && !defined(ZERO)
-  HeapWord* start = (HeapWord*)start_address() + sp() - frame::sender_sp_offset;
+#if (defined(X86) || defined(AARCH64) || defined(RISCV64)) && !defined(ZERO)
+  HeapWord* start = (HeapWord*)start_address() + sp() - frame::sender_sp_offset RISCV64_ONLY(- frame::metadata_words);
```

Comment:
Also `frame::sender_sp_offset` is a platform detail that cannot be used in shared code. Since `metadata_words = sender_sp_offset` on other platforms I'd suggest you replace `frame::sender_sp_offset` with `frame::metadata_words`. Probably this has been forgotten.

Reply:
Yes, that makes sence. Fixed.